transport/nats/encode_decode.go

```diff
@@ -0,0 +1,33 @@
+package nats
+
+import (
+	"context"
+
+	"github.com/nats-io/go-nats"
+)
+
+// DecodeRequestFunc extracts a user-domain request object from a publisher
+// request object. It's designed to be used in NATS subscribers, for subscriber-side
+// endpoints. One straightforward DecodeRequestFunc could be something that
+// JSON decodes from the request body to the concrete response type.
+type DecodeRequestFunc func(context.Context, *nats.Msg) (request interface{}, err error)
+
+
+// EncodeRequestFunc encodes the passed request object into the HTTP request
```

Comment:
This comment still talks about HTTP

Reply:
oh, great! thanks a lot! I didn't hope to so fast review) Actually, it was a draft and I supposed to modify comments and test code. One more time, thanks a lot! I will fix your remarks as soon as possible.